File: 83-RemoveDuplicatesFromSortedList/83-RemoveDuplicatesFromSortedList.java
// Last updated: 24/07/2026, 23:54:08
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode currNode = head;

        while(currNode!=null&&currNode.next != null){
            if(currNode.val == currNode.next.val){
                currNode.next=currNode.next.next;
            }else{
                currNode=currNode.next;
            }  
        }
        return head;
    }
}

File: 83-RemoveDuplicatesFromSortedList/83-RemoveDuplicatesFromSortedList.js
// Last updated: 24/07/2026, 23:54:01
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let temp=head;
    let newNode=head;

    while (temp !== null){
        if(temp.val !== newNode.val){
            newNode.next = temp;
            newNode = temp;
        }

         if(temp.next == null){
            newNode.next = null
        }
        temp=temp.next
    } 

    return head;
};